build.ps1

```diff
@@ -84,124 +86,121 @@ Function CheckPublishTools()
 
 Function CheckTools($commands)
 {
-    foreach($command in $commands)
+    foreach ($command in $commands)
     {
         $info = Get-Command $command -ErrorAction SilentlyContinue
-        if ($info -eq $null)
+        if (-not $info)
         {
             throw "Toolset not found: '$command' is missing."
         }
     }
 }
 
-Function BuildProject($path, $message) {
-
+Function BuildProject($path, $message)
+{
     $label = "BUILD: --- $message $configuration ---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
-    cd (Join-Path $rootDir $path)
 
-    if ($clean) {
-        & dotnet clean --verbosity $verbosity --configuration $configuration
-        if ($LASTEXITCODE -ne 0) {
+    $projectPath = Join-Path $rootDir $path
+
+    if ($clean)
+    {
+        & dotnet clean $projectPath --verbosity $verbosity --configuration $configuration
+        if ($LASTEXITCODE -ne 0)
+        {
             throw "Clean failed: $label"
         }
     }
 
-    & dotnet build --verbosity $verbosity --configuration $configuration
+    & dotnet build $projectPath --verbosity $verbosity --configuration $configuration
 
-    if ($LASTEXITCODE -ne 0) {
+    if ($LASTEXITCODE -ne 0)
+    {
         throw "Build failed: $label"
     }
 }
 
-Function BuildPackage($path, $message) {
+Function BuildPackage($path, $message)
+{
     $label = "PACK: --- $message $configuration ---"
 
     Write-Host
     Write-Host -ForegroundColor Cyan $label
 
     $projectPath = Join-Path $rootDir $path
-    cd $projectPath
-
-    $projectName = (dir (Join-Path $projectPath *.csproj))[0].BaseName
+    $projectName = (Get-ChildItem (Join-Path $projectPath *.csproj))[0].BaseName
 
     if ($sign)
     {
-        Write-Host -ForegroundColor Magenta "`tSigning binaries: $projectName"
-        $filesToSign = dir -Recurse .\bin\Release\$projectName.dll
+        Set-Location $projectPath
+        Write-Host -ForegroundColor Magenta "`tSigning binaries for: $projectName"
+        $filesToSign = Get-ChildItem -Path "$projectPath\bin\$configuration\*\$projectName.dll" -Recurse
```

Comment:
we would not want to sign debug packages; maybe we could add some other validation to check that sign is always called on release only?

Reply:
I put in some extra error handling in the initial $sign check and throw if not Release.